### ISSUE TITLE
Aptos Account address generation and authentication key rotation

### ISSUE THREAD

**Issue by zhaomengru2015** [KeystoneHQ/Keystone-developer-hub#6]

**Comment by zhaomengru2015** [comment]
I think the user case for key rotation is more like a best security practice, like today organizations rotate their credentials periodically. 
For the case "the mnemonic words exported", I think rotate mnemonic words might not solve the problem. If user's mnemonic words was exported, it much likely that the assets has already lost. So in this case, the first thing user should do is transfer the assets to another safer place, not rotate another authentication key with a new mnemonic word.

**Comment by MartianSiddharth** [comment]
I think mnemonic phrase rotation would be more powerful compared to key rotation but I understand the complexity it brings to hardware wallets. Key rotation can be used as a best practice too but providing both these features to users is a big UX problem.

**Comment by MartianSiddharth** [comment]
Based on the discussion from this [thread](https://github.com/aptos-labs/aptos-core/issues/566), Martian wallet will be using a different key rotation technique and the mnemonic phrase rotation will leverage this algorithm. [Aptos Wallet ](https://github.com/aptos-labs/aptos-core/pull/2972) is also using this technique for key rotation.

**Comment by MartianSiddharth** [comment]
I agree with your proposal to incorporate SLIP-10 to use ed25519 curve instead of secp256k1 curve but why are we not using SLIP-10 for master key generation. Can you please elaborate on this part?

> generate the master seed from mnemonic words following the [bip39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) standard.
derive private and public key pair for ed25519 curve for hdPath "m/44'/637'/0'/0'/0'" following the [slip10](https://github.com/satoshilabs/slips/blob/master/slip-0010.md) standard. the sha3-256(pubkey_A | signing_schema) will be the account's initial authentication key according to the [Aptos Authentication Key Definition](https://aptos.dev/concepts/basics-accounts/#authentication-keys), so as the account's address.
more private and public key pairs can be derived by increasing the account level hdPath described in step 2; respect the address gap limit described below.

**Comment by zhaomengru2015** [comment]
> I agree with your proposal to incorporate SLIP-10 to use ed25519 curve instead of secp256k1 curve but why are we not using SLIP-10 for master key generation. Can you please elaborate on this part?
@MartianSiddharth oh we'll definitely use the SLIP10 for master key generation, what we said in the issue is the about obtaining seed bytes from a BIP-0039 mnemonic, should be the same as slip10.

**Comment by zhaomengru2015** [comment]
@MartianSiddharth about the  mnemonic phrase rotation, can you please give more details about the algorithm?

**Comment by zhaomengru2015** [comment]
@MartianSiddharth  as for the SLIP-10 master key generation, my fault, what I mean was definitely using SLIP10 to master key generation, and derive child private and public key pairs for account and rotation keys, let me update the issue description.

**Comment by MartianSiddharth** [comment]
@zhaomengru2015 I had another doubt about SLIP-10. 

> For the ed25519 curve the private keys are no longer multipliers for the group generator; instead the hash of the private key is the multiplier. For this reason, our scheme for ed25519 does not support public key derivation and uses the produced hashes directly as private keys.

Does this mean that we will not be able to derive public keys?

**Comment by zhaomengru2015** [comment]
@MartianSiddharth No，We don't. We can only get privatekey from slip10. Aptos will derive an ed25519 signing key pair with [nacl]（https://www.npmjs.com/package/tweetnacl）

**Comment by zhaomengru2015** [comment]
@MartianSiddharth

**Comment by zhaomengru2015** [comment]
@MartianSiddharth , can we support key rotation for hardware wallet? like when users connect Keystone in Martian, then using key rotation, users select a `derivationPath` and Keystone generate the key.  Using mnemonic rotation if users connect without hardware wallet?

>I think mnemonic phrase rotation would be more powerful compared to key rotation but I understand the complexity it brings to hardware wallets. Key rotation can be used as a best practice too but providing both these features to users is a big UX problem.

**Comment by zhaomengru2015** [comment]
@MartianSiddharth do you have any feedback about the Key rotation suggestions ⬆️ ? Or you still want to go with only mnemonic rotation at this stage, I think that could be fine.

**Comment by aaronisme** [comment]
move this to the official AIP repo https://github.com/aptos-labs/aip/issues/2
